tests/server/test_desired_state_versions.py

```diff
@@ -20,17 +20,112 @@
 import json
 import uuid
 from operator import itemgetter
+from typing import Any, Optional
 
 import pytest
 from tornado.httpclient import AsyncHTTPClient, HTTPRequest
 
 from inmanta import data
-from inmanta.data.model import DesiredStateLabel, PromoteTriggerMethod
-from inmanta.server import config
+from inmanta.agent.config import agent_reconnect_delay, server_timeout
+from inmanta.const import AGENT_SCHEDULER_ID
+from inmanta.data import ConfigurationModel, Scheduler
+from inmanta.data.model import DesiredStateLabel
+from inmanta.protocol import SessionEndpoint, methods
+from inmanta.server import SLICE_AGENT_MANAGER, config, protocol
+from inmanta.types import Apireturn
+from utils import retry_limited
+
+
+class BaseAgent(SessionEndpoint):
+
+    def __init__(
+        self,
+        environment: Optional[uuid.UUID] = None,
+    ):
+        """
+        :param environment: environment id
+        """
+        super().__init__(name="agent", timeout=server_timeout.get(), reconnect_delay=agent_reconnect_delay.get())
+        self._env_id = environment
+        self.enabled: dict[str, bool] = {}
```

Comment:
What is the point of the `enabled` variable? Is it used anywhere?

Reply:
Copy-paste, I removed it